### DISCUSSION TITLE
Refresh Tokens and JWT's

### DISCUSSION THREAD

**Discussion by ethan-davies** [unkeyed/unkey#1871] [Q&A]
I think it would be nice if there was were changes implemented so keys can be set as JWTs and also allow for refresh tokens to be made so expired keys can be regenerated. This may help people implement an OAuth2 system with unkey and overall I think the changes would be beneficial.

**Comment by perkinsjr** [comment]
Hey @ethan-davies 

How does this fit into the Unkey API management platform? Are talking about creating your own OAuth2 provider? Or are you talking about the OIDC which could be used to provide server to server communication?

One of those makes sense for us, an we have looked at, the other is FE auth more than API related and 100s of companies already do that.
